pokemon_battle_arena/src/arena/mod.rs

```diff
@@ -49,10 +52,12 @@ impl Arena {
     // Setter Methods
     //
     /// Sets the effect of the arena
-    pub fn set_effect(&mut self, new: enums::types) {
+    #[allow(dead_code)]
+    pub fn set_effect(&mut self, new: enums::Types) {

```

Comment:
If you have a bunch of `set` functions which do not check the input, you can just mark the field as public. This avoids setter/getter spam. Setter/getter pairs are only useful if the setter executes certain checks to assert consistency.